core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1094,30 +1097,29 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
             case fold: FoldZIO[Any, Any, Any, Any, Any] =>
               updateLastTrace(fold.trace)
 
-              stackIndex = pushStackFrame(fold, stackIndex)
+              @inline def exitFailure(cause: Cause[Any]): ZIO.Erased =
+                if (shouldInterrupt()) Exit.Failure(cause.stripFailures) else fold.failureK(cause)
```

Comment:
This comment is redundant because I think we'll end up removing this code, but as a side-note, the `@inline` annotation doesn't work for Scala 3. So for performance-critical code paths we either shouldn't be using it or we should make it Scala-version specific

Reply:
Removed